src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2379,23 +2406,16 @@ private bool ConsumeSingleLineCommentMultiSegment(ReadOnlySpan<byte> localBuffer
             _lineNumber++;
 
         Done:
-            if (HasValueSequence)
+            ReadOnlySequence<byte> commentSequence = _sequence.Slice(start, end);
```

Comment:
Yeah there could be a way. But that was getting messier to track two distinct behaviors as in `indexes` in case of single segment and `start / end` in case of segments within one single method. In fact I did this reduction with thought that anyways single segment is a special case of multi-segments (which could be applied to other places as well?).

Curiosity question, is working with sequences avoidable due to reasons such as performance, which is why this question?

Reply:
> Curiosity question, is working with sequences avoidable due to reasons such as performance, which is why this question?

Yes. Generally, we want to avoid slicing a readonlysequence if possible. In this case, if it introduces too much complexity, it's fine leaving it as is (since this slice overload is not too expensive).

Reply:
Thanks! Leaving it as is.

Reply:
I have reworked this entirely as we discussed. Please compare code changes with original unmodified code in master.